### PR TITLE
tests: Test tag cleanup

### PR DESCRIPTION
**Pull request by teburd**
Several test tags were inconsistently upper cased, some tests were missing appropriate tags that could be used to identify and exclude certain tests. This PR fixes several issues found while determining tags to ignore for intel platforms. While the ignore_tags updates were decided to not be the approach to use, the tag fixes are still useful.

### CODE REVIEW

**Review by andyross — DISMISSED** [review]

File: boards/xtensa/intel_adsp_ace15_mtpm/intel_adsp_ace15_mtpm.yaml [hunk 1]
```diff
@@ -9,5 +9,34 @@ supported:
   - dma
 testing:
   ignore_tags:
+    - cmsis_dsp
+    - cmsis_rtos
+    - zdsp
```
Comment (dcpleung) [inline boards/xtensa/intel_adsp_ace15_mtpm/intel_adsp_ace15_mtpm.yaml:14]: `zdsp` is duplicated.

Comment (teburd) [reply boards/xtensa/intel_adsp_ace15_mtpm/intel_adsp_ace15_mtpm.yaml:14]: I wanted to doubly ensure it was ignored

**Comment by teburd** [comment]
Split off the cmake rimage target generation as its quite useful and not really related to the tag fixups/ignore lists.

https://github.com/zephyrproject-rtos/zephyr/pull/52942

**Review by dcpleung — DISMISSED** [review]

File: boards/xtensa/intel_adsp_cavs15/intel_adsp_cavs15.yaml [hunk 2]
```diff
@@ -5,7 +5,37 @@ arch: xtensa
...
+    - sensor
+    - sensors
+    - disk
+    - led
+    - zbus
+    - chre
+    - nanopb
+    - smf
+    - compression
+    - dmic
```
Comment (galak) [inline boards/xtensa/intel_adsp_cavs15/intel_adsp_cavs15.yaml:34]: should we be ignoring dmic?

File: samples/drivers/i2s/echo/sample.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@ sample:
   name: I2S echo sample
 tests:
   sample.drivers.i2s.echo:
-    tags: I2S
+    tags: I2S i2s
```
Comment (galak) [inline samples/drivers/i2s/echo/sample.yaml:5]: just remove the upper case I2S tag since its not used.

**Review by galak — DISMISSED** [review]
drop uppercase I2S tag.

File: boards/xtensa/intel_adsp_cavs15/intel_adsp_cavs15.yaml (hunk 2 above)
Comment (teburd) [reply boards/xtensa/intel_adsp_cavs15/intel_adsp_cavs15.yaml:34]: yes, in sof (and for intel) we only care about the dai peripheral interface for all things audio, dmic isn't used

**Review by dcpleung — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Comment by MaureenHelm** [comment]
Needs rebase

**Review by dcpleung — DISMISSED** [review]

**Review by nashif — CHANGES_REQUESTED** [review]

File: boards/xtensa/intel_adsp_ace15_mtpm/intel_adsp_ace15_mtpm.yaml [hunk 4]
```diff
@@ -9,5 +9,33 @@ supported:
   - dma
 testing:
   ignore_tags:
```
[the remark this answers is not in the document]
Comment (teburd) [reply boards/xtensa/intel_adsp_ace15_mtpm/intel_adsp_ace15_mtpm.yaml]: Removed the changes to the ignore tags

**Comment by teburd** [comment]
Update 1:

 * Removes the ignore_tags changes to the intel platforms
 * Keeps all the tag changes
 * Renames PR and updates comment

**Review by dcpleung — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

**Pull request merged**